core/src/main/java/com/graphhopper/routing/util/parsers/OSMHgvParser.java

```diff
@@ -5,22 +5,29 @@
 import com.graphhopper.routing.ev.EdgeIntAccess;
 import com.graphhopper.routing.ev.EnumEncodedValue;
 import com.graphhopper.routing.ev.Hgv;
+import com.graphhopper.routing.util.TransportationMode;
 import com.graphhopper.storage.IntsRef;
 
 import static com.graphhopper.routing.util.parsers.helpers.OSMValueExtractor.conditionalWeightToTons;
 
-public class OSMHgvParser implements TagParser {
-    EnumEncodedValue<Hgv> hgvEnc;
+public class OSMHgvParser extends OSMRoadAccessParser<Hgv> {
 
     public OSMHgvParser(EnumEncodedValue<Hgv> hgvEnc) {
-        this.hgvEnc = hgvEnc;
+        super(hgvEnc, OSMRoadAccessParser.toOSMRestrictions(TransportationMode.HGV),
+                (readerWay, accessValue) -> accessValue, Hgv::find);
     }
 
     @Override
     public void handleWayTags(int edgeId, EdgeIntAccess edgeIntAccess, ReaderWay way, IntsRef relationFlags) {
         String value = way.getTag("hgv:conditional", "");
         int index = value.indexOf("@");
-        Hgv hgvValue = index > 0 && conditionalWeightToTons(value) == 3.5 ? Hgv.find(value.substring(0, index).trim()) : Hgv.find(way.getTag("hgv"));
-        hgvEnc.setEnum(false, edgeId, edgeIntAccess, hgvValue);
+        if (index > 0 && conditionalWeightToTons(value) == 3.5) {
```

Comment:
Why do we check for *exactly* 3.5t?

The following examples wouldn't work while they seem perfectly fine IMHO:

* `private @ (weight>12)`
* `delivery @ (weight>5.5)`

```suggestion
        if (index > 0 && conditionalWeightToTons(value) >= 3.5) {
```